app/uk/gov/hmrc/mobileversioncheck/controllers/VersionCheckController.scala

```diff
@@ -76,14 +84,40 @@ class SandboxVersionCheckController @Inject()(
     with VersionCheckController {
   override def parser: BodyParser[AnyContent] = cc.parsers.anyContent
 
-  override def doVersionCheck(deviceVersion: DeviceVersion, journeyId: String)(
+  override def doVersionCheck(deviceVersion: DeviceVersion, journeyId: String, callingService: String)(
     implicit hc:                             HeaderCarrier,
     request:                                 Request[_]): Future[Result] = {
 
     val result: Result = request.headers.get("SANDBOX-CONTROL") match {
-      case Some("ERROR-500")        => InternalServerError
-      case Some("UPGRADE-REQUIRED") => Ok(Json.toJson(PreFlightCheckResponse(true)))
-      case _                        => Ok(Json.toJson(PreFlightCheckResponse(false)))
+      case Some("ERROR-500") => InternalServerError
+      case Some("UPGRADE-REQUIRED") =>
+        Ok(
+          Json.toJson(
+            PreFlightCheckResponse(
+              upgradeRequired = true,
+              AppState(ACTIVE, None)
+            )))
+      case Some("INACTIVE-APPSTATE") =>
+        Ok(
+          Json.toJson(
+            PreFlightCheckResponse(
+              upgradeRequired = false,
+              AppState(INACTIVE, Some(LocalDateTime.parse("2019-11-01T00:00:00", DateTimeFormatter.ISO_LOCAL_DATE_TIME)))
+            )))
+      case Some("SHUTTERED-APPSTATE") =>
+        Ok(
+          Json.toJson(
+            PreFlightCheckResponse(
+              upgradeRequired = false,
+              AppState(SHUTTERED, Some(LocalDateTime.parse("2020-01-01T00:00:00", DateTimeFormatter.ISO_LOCAL_DATE_TIME)))
+            )))
+      case _ =>
+        Ok(
+          Json.toJson(
+            PreFlightCheckResponse(
+              upgradeRequired = false,
+              AppState(ACTIVE, None)
+            )))
     }
```

Comment:
Might need to have a check on the sandbox to make sure that these do not return if not RDS